src/logic/function/basic.lean

```diff
@@ -357,6 +357,22 @@ rfl
 
 end bicomp
 
+section uncurry
+
+variables {α β γ δ : Type*}
+
+/-- Records a way to turn an element of `α` into a function from `β` to `γ`. -/
```

Comment:
It feels like somewhere there should be a docstring explaining that this applies repeatedly, maybe here, maybe on the notation.